From 99d9dd107745d0f42b4d72507a1edeb42ecbb637 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 2 Dec 2014 21:03:28 -0500
Subject: [x86] Use new KABI macros

Message-id: <1417554215-238434-4-git-send-email-dzickus@redhat.com>
Patchwork-id: 101090
O-Subject: [RHEL7 PATCH 03/10] kabi, x86: Use new KABI macros
Bugzilla: 1164383
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1164383
Upstream: RHEL only

Convert some of the x86 GENKSYMS usage to use the new kabi macros.

Reviewed-by: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Don Zickus <dzickus@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/processor.h b/arch/x86/include/asm/processor.h
index 5604098..37db43e 100644
--- a/arch/x86/include/asm/processor.h
+++ b/arch/x86/include/asm/processor.h
@@ -30,6 +30,8 @@ struct mm_struct;
 #include <linux/err.h>
 #include <linux/irqflags.h>
 
+#include <linux/rh_kabi.h>
+
 /*
  * We handle most unaligned accesses in hardware.  On the other hand
  * unaligned DMA can be quite expensive on some Nehalem processors.
@@ -404,12 +406,10 @@ struct xsave_struct {
  struct i387_fxsave_struct i387;
  struct xsave_hdr_struct xsave_hdr;
  struct ymmh_struct ymmh;
-#ifndef __GENKSYMS__
- struct lwp_struct lwp;
- struct bndregs_struct bndregs;
- struct bndcsr_struct bndcsr;
+ RH_KABI_EXTEND(struct lwp_struct lwp)
+ RH_KABI_EXTEND(struct bndregs_struct bndregs)
+ RH_KABI_EXTEND(struct bndcsr_struct bndcsr)
  /* new processor state extensions will go here */
-#endif
 } __attribute__ ((packed, aligned (64)));
 
 union thread_xstate {
-- 
1.7.1